cmds/clients/contestcli-http/main.go

```diff
@@ -149,12 +183,40 @@ func run(verb string) error {
 		encoder.SetIndent("", " ")
 		err := encoder.Encode(apiErr)
 		if err != nil {
-			return fmt.Errorf("cannot re-encode httplistener.HTTPAPIResponse object: %v", err)
+			return "", fmt.Errorf("cannot re-encode httplistener.HTTPAPIResponse object: %v", err)
 		}
 		indentedJSON = buffer.Bytes()
 	}
 	// this is the only thing we want on stdout - the JSON-formatted response,
 	// so it can be piped to other tools if desired.
-	fmt.Println(string(indentedJSON))
-	return nil
+	return string(indentedJSON), nil
 }
+ 
+func wait(params url.Values) (string, error) {
+	
+	var finished bool
+	time.Sleep(jobWaitPoll)
```

Comment:
the sleep could go at the end, so that long intervals would not delay the first check too much, and it's probably desirable to have the first check shortly after the job submission to know before 10 seconds whether the job failed immediately or not

Reply:
Done, but the down side of this is an extra useless call for the tests that I'm doing since the jobs don't complete that fast. However, I'm not familiar with the system so maybe some failures can be fast enough to make use of that.

Reply:
I'd say it's better to have one more call rather than having to wait a possibly too long time before the first check. Otherwise, the first check could have a short delay, e.g. 1s-3s